docs/topics/whatsnew-eap.md

```diff
@@ -87,11 +84,153 @@ We would appreciate any feedback you may have!
+        compilations.getByName("main") {
+            cinterops {
+                val cinterop by creating {
+                    definitionFile.set(project.file("def-file.def"))
+                }
+            }
+        }
+    }
+}
+```
+
+</tab>
+
+<tab id="groovy" title="Groovy">
+
+```groovy
+kotlin {
+    macosArm64("native") {
+        compilations.main {
+            cinterops {
+                cinterop {
+                    definitionFile.set(project.file("def-file.def"))
+                }
+            }
+        }
+    }
+}
+```
+
+</tab>
+
+</tabs>
+
+> `defFile` and `defFileProperty` parameters are now deprecated.
+>
+{type="warning"}
+
+## Visibility changes in Gradle
+
+> This change doesn't impact Groovy DSL users.
+> 
+{type="note"}
+
+In Kotlin %kotlinEapVersion%, we've modified the Kotlin Gradle Plugin for better control and safety in your build scripts.
+Previously, certain Kotlin DSL functions and properties intended for a specific DSL context would inadvertently leak to 
+other DSL contexts. This leakage could lead to the use of incorrect compiler options, settings being applied multiple times,
+and other misconfigurations:
+
+```kotlin
+kotlin {
+  // Couldn't access methods and properties defined in the 
+  // extension class
+    jvm {
+  // Couldn't access methods and properties defined in the
+  // extension and Kotlin target classes
+        compilations.configureEach {
+  // Couldn't access methods and properties defined in the
+  // extension, Kotlin target or Kotlin compilation classes
+            compileTaskProvider.configure {
+              
+    // For example:
+                explicitApi() 
+                // ERROR as it is defined in the extension class
+                mavenPublication {} 
+                // ERROR as it is defined in the Kotlin target class
+                defaultSourceSet {} 
+                // ERROR as it is defined in the Kotlin compilation class
+            }
+        }
+    }
+}
+```
+
+To fix this issue, we've added the `@KotlinGradlePluginDsl` annotation,
+preventing the exposure of the Kotlin Gradle plugin DSL functions and properties to the levels where they are not intended
+to be available. The following levels are separated from each other:
+
+* Kotlin extension
+* Kotlin target
+* Kotlin compilation
+* Kotlin compilation task
+
+If your build script is configured incorrectly, you should see compiler warnings with suggestions on how to fix it. For example:
+
+```kotlin
+kotlin {
+    jvm {
+        sourceSets.getByName("jvmMain").dependencies {
+            implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:1.7.3")
+        }
+    }
+}
+```
+
+In this case, the warning message for `sourceSets` is:
+
+```kotlin
+[DEPRECATION] 'sourceSets: NamedDomainObjectContainer<KotlinSourceSet>' is deprecated.Accessing 'sourceSets' container on the Kotlin target level DSL is deprecated . Consider configuring 'sourceSets' on the Kotlin extension level .
```

Comment:
Why are there whitespaces around full stops?

Reply:
Bug in the error message 😞